src/overlay005/field_control.c

```diff
@@ -0,0 +1,1092 @@
+    input->unused4 = FALSE;
+    input->unused5 = FALSE;
+    input->playerDir = DIR_NONE;
+    input->transitionDir = DIR_NONE;
+}
+
+void FieldInput_Update (FieldInput *input, FieldSystem *fieldSystem, u16 pressedKeys, u16 heldKeys)
+{
+    FieldInput_Clear(input);
+
+    int moveState = Player_MoveState(fieldSystem->playerAvatar);
+    int avatarState = PlayerAvatar_MoveState(fieldSystem->playerAvatar);
+    int playerDir = PlayerAvatar_GetDir(fieldSystem->playerAvatar);
+
+    input->pressedKeys = pressedKeys;
+    input->heldKeys = heldKeys;
+
+    if (moveState == PLAYER_MOVE_STATE_END || moveState == PLAYER_MOVE_STATE_NONE) {
+        if (pressedKeys & PAD_BUTTON_X) {
+            input->menu = TRUE;
+        }
+
+        if (pressedKeys & PAD_BUTTON_Y) {
+            input->registeredItem = TRUE;
+        }
+
+        if (pressedKeys & PAD_BUTTON_A) {
+            input->interact = TRUE;
+        }
+
+        if (pressedKeys & PAD_BUTTON_B) {
+            input->unused1 = TRUE;
+        }
+
+        if (heldKeys & (PAD_KEY_UP | PAD_KEY_DOWN | PAD_KEY_LEFT | PAD_KEY_RIGHT)) {
+            input->sign = TRUE;
+        }
+
+        if (heldKeys & (PAD_KEY_UP | PAD_KEY_DOWN | PAD_KEY_LEFT | PAD_KEY_RIGHT)) {
+            input->mapTransition = TRUE;
+        }
+    }
+
+    if (moveState == PLAYER_MOVE_STATE_END && avatarState == AVATAR_MOVE_STATE_MOVING) {
+        input->movement = TRUE;
+    }
+
+    if (moveState == PLAYER_MOVE_STATE_END) {
+        input->endMovement = TRUE;
+    }
+
+    if (playerDir == DIR_NORTH && heldKeys & PAD_KEY_UP || 
+        playerDir == DIR_SOUTH && heldKeys & PAD_KEY_DOWN || 
+        playerDir == DIR_WEST  && heldKeys & PAD_KEY_LEFT || 
+        playerDir == DIR_EAST  && heldKeys & PAD_KEY_RIGHT) {
+
+        input->transitionDir = playerDir;
+    } else {
+        input->transitionDir = DIR_NONE;
+    }
+
+    input->playerDir = sub_02061308(fieldSystem->playerAvatar, pressedKeys, heldKeys);
+}
+
+int FieldInput_Process (const FieldInput *input, FieldSystem *fieldSystem)
+{
+    if (input->unused5 == FALSE && sub_0203F5C0(fieldSystem, 1) == TRUE) {
+        return TRUE;
+    }
+
+    if (input->unused5 == FALSE) {
+        BOOL doublesCheck = sub_02054AB0(Party_GetFromSavedata(fieldSystem->saveData));
+
+        if (sub_0206A984(SaveData_Events(fieldSystem->saveData)) == TRUE) {
+            doublesCheck = TRUE;
+        }
+
+        if (sub_02067A84(fieldSystem, doublesCheck) == TRUE || 
+            sub_02071CB4(fieldSystem, 2) == TRUE && 
+            ov8_0224C51C(fieldSystem) == TRUE) {
+
+            sub_0205F56C(fieldSystem->playerAvatar);
+            MapObjectMan_PauseAllMovement(fieldSystem->mapObjMan);
+            return TRUE;
+        }
+    }
+
+    if (input->movement) {
+        sub_0206A9A4(SaveData_Events(fieldSystem->saveData));
+
+        if (Field_ProcessStep(fieldSystem) == TRUE) {
+            return TRUE;
+        }
+    }
+
+    if (input->unused5 == FALSE) {
```

Comment:
Is this branch always taken?

Reply:
I would imagine yes, but it doesn't match if I rip out that `if` statement.

Reply:
That makes sense to me; the compiler likely doesn't have any insight into the possibility of the memory slot being edited elsewhere.

Reply:
Then maybe leave a comment saying so (and also this field probably shouldn't be named `unused`, since it is used for comparisons in here and in other places)